src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionService.cs

```diff
@@ -40,10 +40,31 @@ internal abstract class AbstractGoToDefinitionService : AbstractFindDefinitionSe
         async Task<IEnumerable<INavigableItem>?> IGoToDefinitionService.FindDefinitionsAsync(Document document, int position, CancellationToken cancellationToken)
             => await FindDefinitionsAsync(document, position, cancellationToken).ConfigureAwait(false);
 
+        private async Task<bool> TryNavigateToSpanAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            // Can only navigate or present items on UI thread.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
did you copyu this from some other area?

Reply:
Yes, see `TryNavigateToOrPresentItemsAsync`. Honestly I have no idea, so tried to follow pre-existing pattern :-P